src/utils/createMicrotask/createMicrotask.ts

```diff
@@ -0,0 +1,8 @@
+/**
+ * Creates a promise that resolves at the end of the current Task.
+ */
+export function createMicrotask(): Promise<void> {
+  return new Promise((resolve) => {
+    queueMicrotask(resolve);
+  });
```

Comment:
This seems a bit redundant based on the fact that promises already use microtasks internally as part of resolving.

https://developer.mozilla.org/en-US/docs/Web/API/HTML_DOM_API/Microtask_guide

> JavaScript [promises](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise) and the [Mutation Observer API](https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver) both use the microtask queue to run their callbacks, but there are other times when the ability to defer work until the current event loop pass is wrapping up. In order to allow microtasks to be used by third-party libraries, frameworks, and polyfills, the [queueMicrotask()](https://developer.mozilla.org/en-US/docs/Web/API/queueMicrotask) method is exposed on the [Window](https://developer.mozilla.org/en-US/docs/Web/API/Window) and [Worker](https://developer.mozilla.org/en-US/docs/Web/API/Worker) interfaces.

https://javascript.info/microtask-queue

> Asynchronous tasks need proper management. For that, the ECMA standard specifies an internal queue PromiseJobs, more often referred to as the “microtask queue” (V8 term).
> 
> As stated in the [specification](https://tc39.github.io/ecma262/#sec-jobs-and-job-queues):
> 
> The queue is first-in-first-out: tasks enqueued first are run first.
> Execution of a task is initiated only when nothing else is running.
> Or, to put it more simply, when a promise is ready, its .then/catch/finally handlers are put into the queue; they are not executed yet. When the JavaScript engine becomes free from the current code, it takes a task from the queue and executes it.

So, these things are "exactly" the same:

```js
await undefined;
await Promise.resolve();
await createMicrotask();
```

Or potentially, wrapping the microtask in a promise messes up the FIFO ordering, since it's queued twice, making that solution worse.

Reply:
Yeah thought of that as well, this is more explicit/similar to `createTimeout`. But I like the approach of just returning a resolved promise.